### PR TITLE
Wconversion enabler

### PR DESCRIPTION
**Pull request by mgaio**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/272%23issuecomment-146456162%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/272%23issuecomment-147700831%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/272%23discussion_r41874505%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/272%23discussion_r41878765%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/272%23issuecomment-146456162%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%40krocard%20Please%20review%22%2C%20%22created_at%22%3A%20%222015-10-08T08%3A29%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-13T12%3A28%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c3bd9728f9848835226c9ed317144b1a19430c18%20parameter/BitParameterType.cpp%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/272%23discussion_r41874505%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20a%20real%20bug%2C%20bitParameter%20parameter%20may%20be%2064%20bit.%22%2C%20%22created_at%22%3A%20%222015-10-13T14%3A42%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yup%2C%20already%20detected%20and%20filed.%20Unfortunately%2C%20the%20bug%20is%20in%20the%20API.%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A09%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/BitParameterType.cpp%3AL182-191%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/272#issuecomment-146456162'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> :+1: @krocard Please review
- <a href='https://github.com/dawagner'></a> @krocard please review
- [x] <a href='#crh-comment-Pull c3bd9728f9848835226c9ed317144b1a19430c18 parameter/BitParameterType.cpp 8'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/272#discussion_r41874505'>File: parameter/BitParameterType.cpp:L182-191</a></b>
- <a href='https://github.com/krocard'></a> This is a real bug, bitParameter parameter may be 64 bit.
- <a href='https://github.com/dawagner'></a> Yup, already detected and filed. Unfortunately, the bug is in the API.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/272?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/272?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/272'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/272'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
:+1: @krocard Please review

**Comment by dawagner** [comment]
@krocard please review

File: parameter/BitParameterType.cpp [hunk 1]
```diff
@@ -182,9 +182,9 @@ bool CBitParameterType::toBlackboard(uint64_t uiUserValue, uint64_t& uiValue, CP
     return true;
 }
 
-void CBitParameterType::fromBlackboard(uint32_t& uiUserValue, uint64_t uiValue, CParameterAccessContext& /*ctx*/) const
+void CBitParameterType::fromBlackboard(uint32_t& userValue, uint64_t value, CParameterAccessContext& /*ctx*/) const
 {
-    uiUserValue = (uiValue & getMask()) >> _bitPos;
+    userValue = static_cast<uint32_t>((value & getMask()) >> _bitPos);
```
Comment (krocard) [inline parameter/BitParameterType.cpp]: This is a real bug, bitParameter parameter may be 64 bit.

Comment (dawagner) [inline parameter/BitParameterType.cpp]: Yup, already detected and filed. Unfortunately, the bug is in the API.

**Comment by dawagner** [comment]
taken over at #285

**Comment by krocard** [comment]
:+1: resubmited

**Pull request closed without merging**